auth/api/iam/openid4vp.go

```diff
@@ -513,45 +528,66 @@ func (r Wrapper) validatePresentationNonce(presentations []vc.VerifiablePresenta
 	return returnErr
 }
 
-func (r Wrapper) handleAccessTokenRequest(ctx context.Context, verifier did.DID, authorizationCode *string, redirectURI *string, clientId *string) (HandleTokenRequestResponseObject, error) {
-	// first check redirectURI
-	if redirectURI == nil {
+func (r Wrapper) handleAccessTokenRequest(ctx context.Context, verifier did.DID, request HandleTokenRequestFormdataRequestBody) (HandleTokenRequestResponseObject, error) {
+	// check if redirectURI is present
+	if request.RedirectUri == nil {
 		return nil, oauthError(oauth.InvalidRequest, "missing redirect_uri parameter")
 	}
-	callbackURI, err := url.Parse(*redirectURI)
+	// check if code is present
+	if request.Code == nil {
+		return nil, oauthError(oauth.InvalidRequest, "missing code parameter")
+	}
+	// check if client state is present
+	if request.State == nil {
+		return nil, oauthError(oauth.InvalidRequest, "missing state parameter")
+	}
```

Comment:
The authorization code flow does not contain a `state` param in the access token request (unless added by a different RFC than RFC6749). It is not needed in backchannel communication.